### PR TITLE
All external links to _blank

### PR DESCRIPTION
**Pull request by ziriuz84**
#### Summary
Now all external links have target="_blank" automatically (via js)

#### Ticket Link
https://github.com/mattermost/mattermost-developer-documentation/issues/1142

### CODE REVIEW

**Comment by cwarnermm** [comment]
@ziriuz84 - You'll need to sign the CLA in order for this PR to move forward. Once you do, I can assign you to https://github.com/mattermost/mattermost-developer-documentation/issues/1142 as well. Thanks!

**Comment by ziriuz84** [comment]
> @ziriuz84 - You'll need to sign the CLA in order for this PR to move forward. Once you do, I can assign you to #1142 as well. Thanks!

I've just done it just minutes after PR, when the bot asked it in this PR

**Review by neflyte — CHANGES_REQUESTED** [review]

File: site/static/js/main.js [hunk 1]
```diff
@@ -1,53 +1,65 @@
...
+    document.getElementById('navigation').classList.toggle('active');
+  });
+
+  //All external links to target=_blank
+  const content = document.querySelector('body');
+  const links = content.getElementsByTagName("a");
+
+  for (let link of links) {
+    if (link.hostname != window.location.hostname) {
+      link.target = '_blank';
```
[the remark this answers is not in the document]
Comment (ziriuz84) [reply site/static/js/main.js:60]: A shortcode needs to change all the past anchor tags and to remember to use the shortcode in future. I thought that using using js introduces automation in adding _blank target to them, for the past and for the future. If it is needed to use shortcodes and to change them in all the site I will use that, but it must be done manually also in the future for new anchor tags.

[the remark this answers is not in the document]
Comment (neflyte) [reply site/static/js/main.js:60]: That's correct. We're using the `newtabref` shortcode for any link that takes the reader to a different site, but not for links that take the reader to a different page on the docs site.

Thanks!

[the remark this answers is not in the document]
Comment (cwarnermm) [reply site/static/js/main.js:60]: @ziriuz84 - Are you open to updating all external links across the developer documentation site to use the ``newtabref`` shortcode?

[the remark this answers is not in the document]
Comment (ziriuz84) [reply site/static/js/main.js:60]: If you prefer the use of shortcode I can work on it this weekend

**Comment by ziriuz84** [comment]
I'm rewriting all links in blog's articles. I encountered this code 
`[Mattermost JSC vs Herms](https://youtu.be/EnefPsVRSn4)`

Is it ok to rewrite as this?
`{{< newtabref href="https://youtu.be/EnefPsVRSn4" title="Mattermost JSC vs Herms" >}}`

**Comment by neflyte** [comment]
@ziriuz84 Don't worry about the pages in the `blog` or `internal` sections. :slightly_smiling_face:

**Comment by ziriuz84** [comment]
> @ziriuz84 Don't worry about the pages in the `blog` or `internal` sections. slightly_smiling_face They can be ignored.

The issue doesn't specify what parts must be changed, so I started from the beginning :sweat_smile:. What are the specified parts that must be changed?

**Comment by neflyte** [comment]
Only the `integrate` and `contribute` sections are live on the docs site. Sorry for not making that more clear.

**Comment by neflyte** [comment]
Looks like there's a build error:
```
Error: Error building site: "/home/runner/work/mattermost-developer-documentation/mattermost-developer-documentation/site/content/contribute/more-info/webapp/_index.md:11:183": got positional parameter 'Redux'. Cannot mix named and positional parameters
```

Please take a look at the `/contribute/more-info/webapp/_index.md` file and make sure it's using correct shortcode syntax.

**Comment by ziriuz84** [comment]
I corrected the typo in that file. Sorry for the inconvenience.

I have a problem in another file, `/integrate/plugins/interactive-messages/_index.md`

At line 379 with a wrong shortcode like this
`{{< newtabref href="" title="https://docs.mattermost.com/configure/configuration-settings.html#allow-untrusted-internal-connections-to"AllowedUntrustedInternalConnections" config.json setting" >}}`
I encounter this arror, and it's ok
```
❯ make run
hugo server --buildDrafts --disableFastRender -F -s site
Built in 119 ms
Error: Error building site: "/home/sirio/side/mattermost-developer-documentation/site/content/integrate/plugins/interactive-messages/_index.md:380:282": got positional parameter 'AllowedUntrustedInternalConnections'. Cannot mix named and positional parameters
make: *** [Makefile:24: run] Error 255
```

I tried a lot of solution to repair it, also old version in plain markdown and I retrieve this type of error
```

❯ make run                 
hugo server --buildDrafts --disableFastRender -F -s site
Building sites … panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xfae6d8]

goroutine 130 [running]:
github.com/gohugoio/hugo/hugolib.(*shortcodeHandler).extractShortcode(0xc003be4a80, 0x9, 0x0, 0xc003bcfd20, 0x3, 0x4, 0x18bbb20)
	github.com/gohugoio/hugo/hugolib/shortcode.go:505 +0x898
github.com/gohugoio/hugo/hugolib.(*pageState).mapContent(0xc003bedda0, 0xc003570a00, 0xc003708c80, 0xc001f06cc0, 0x2114840)
	github.com/gohugoio/hugo/hugolib/page.go:783 +0xb96
github.com/gohugoio/hugo/hugolib.(*pageMap).newPageFromContentNode(0xc0007fe4d0, 0xc0008e1770, 0xc003570a00, 0x0, 0x0, 0x0, 0x0)
	github.com/gohugoio/hugo/hugolib/content_map_page.go:164 +0x652
github.com/gohugoio/hugo/hugolib.(*pageMap).assemblePages.func1(0xc0003ecbc0, 0x3a, 0x19a2b20, 0xc0008e1770, 0xc003bd9d00)
	github.com/gohugoio/hugo/hugolib/content_map_page.go:358 +0x16d
github.com/armon/go-radix.recursiveWalk(0xc0008e1950, 0xc003bd9df0, 0x19a2b00)
	github.com/armon/go-radix/radix.go:519 +0xf0
github.com/armon/go-radix.recursiveWalk(0xc0008e1860, 0xc003bd9df0, 0x19a2b00)
	github.com/armon/go-radix/radix.go:525 +0x7e
github.com/armon/go-radix.recursiveWalk(0xc0008e0ed0, 0xc003bd9df0, 0xc52000)
	github.com/armon/go-radix/radix.go:525 +0x7e
github.com/armon/go-radix.recursiveWalk(0xc000985e90, 0xc003bd9df0, 0xc52000)
	github.com/armon/go-radix/radix.go:525 +0x7e
github.com/armon/go-radix.recursiveWalk(0xc0003ffce0, 0xc003bd9df0, 0xf88600)
	github.com/armon/go-radix/radix.go:525 +0x7e
github.com/armon/go-radix.recursiveWalk(0xc00087a600, 0xc003bd9df0, 0x0)
	github.com/armon/go-radix/radix.go:525 +0x7e
github.com/armon/go-radix.recursiveWalk(0xc0004f5aa0, 0xc003bd9df0, 0x0)
	github.com/armon/go-radix/radix.go:525 +0x7e
github.com/armon/go-radix.(*Tree).Walk(...)
	github.com/armon/go-radix/radix.go:447
github.com/gohugoio/hugo/hugolib.(*pageMap).assemblePages(0xc0007fe4d0, 0x0, 0x0)
	github.com/gohugoio/hugo/hugolib/content_map_page.go:331 +0xec
github.com/gohugoio/hugo/hugolib.(*pageMaps).AssemblePages.func1(0xc0007fe4d0, 0x0, 0x0)
	github.com/gohugoio/hugo/hugolib/content_map_page.go:721 +0x68
github.com/gohugoio/hugo/hugolib.(*pageMaps).withMaps.func1(0x0, 0x0)
	github.com/gohugoio/hugo/hugolib/content_map_page.go:788 +0x2e
github.com/gohugoio/hugo/common/para.(*errGroupRunner).Run.func1(0xc00046ff68, 0x0)
	github.com/gohugoio/hugo/common/para/para.go:52 +0x2f
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0034cea50, 0xc000a139e0)
	golang.org/x/sync/errgroup/errgroup.go:57 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/errgroup.go:54 +0x66
make: *** [Makefile:24: run] Error 2
```

Do you know how to investigate it?
Thanks for all the help

**Comment by neflyte** [comment]
@ziriuz84 I see that you have additional double-quotes (`"`) in the `title` argument. You will need to escape them. I think a backslash (`\`) will work for escaping.

**Comment by ziriuz84** [comment]
I tried to use &quot; and to delete directly the line, but the error remains. I don't understand where is the error. I don't think it's a syntax error, it seems a problem with a library

**Comment by neflyte** [comment]
I think the shortcode above isn't specified correctly.

`{{< newtabref href="" title="https://docs.mattermost.com/configure/configuration-settings.html#allow-untrusted-internal-connections-to"AllowedUntrustedInternalConnections" config.json setting" >}}`

...should be ...

`{{<newtabref href="https://docs.mattermost.com/configure/configuration-settings.html#allow-untrusted-internal-connections-to" title="AllowedUntrustedInternalConnections config.json setting">}}`

Give that one a try. :)

**Comment by cwarnermm** [comment]
@ziriuz84 - Are you open to addressing the merge conflicts in this PR?

**Comment by ziriuz84** [comment]
I'm working to resolve a problem probably introduced with my code, maybe related with this conflicts.

**Comment by ziriuz84** [comment]
I completed the porting of external links to newtabref shortcode

File: site/static/js/main.js [hunk 2]
```diff
@@ -1,53 +1,65 @@
...
+    }, false);
+  });
+
+  clickTouch(hamburger, () => {
+    hamburger.classList.toggle('is-active');
+    document.body.classList.toggle('nav-open');
+    document.getElementById('navigation').classList.toggle('active');
+  });
+
+  // //All external links to target=_blank
```
Comment (neflyte) [inline site/static/js/main.js:54]: Please revert this file if there were no changes made to it.

File: site/content/integrate/apps/authentication/oauth2/_index.md [hunk 3]
```diff
@@ -24,9 +24,9 @@ OAuth 2.0 applications are off by default and can be enabled by the System Admin
...
+3. Set {{< newtabref href="https://docs.mattermost.com/administration/config-settings.html#enable-oauth-2-0-service-provider" title="Enable OAuth 2.0 Service Provider" >}} to **True**.
+4. (Optional) If you'd like to allow external applications to post with customizable usernames and profile pictures, then set {{< newtabref href="https://docs.mattermost.com/configure/configuration-settings.html#enable-integrations-to-override-usernames" title="Enable integrations to override usernames" >}} and {{< newtabref href="https://docs.mattermost.com/configure/configuration-settings.html#enable-integrations-to-override-profile-picture-icons" title="Enable integrations to override profile picture icons" >}} to **true**.
+5. (Optional) If you'd like to allow users on your system who are not System Admins to create OAuth 2.0 applications, then set {{< newtabref href="https://docs.mattermost.com/administration/config-settings.html#restrict-managing-integrations-to-adminsRestrict managing integrations to Admins" title="" >}} to **False**.
```
Comment (neflyte) [inline site/content/integrate/apps/authentication/oauth2/_index.md:29]: This link looks incorrect. I think the link title was pasted in the incorrect place.

File: site/content/contribute/more-info/team_contributions/_index.md [hunk 4]
```diff
@@ -8,7 +8,7 @@ aliases:
 
 Mattermost is an open source Slack-alternative that helps teams stay in sync, with messaging and file sharing across PCs and phones, with archiving and search, and a growing community of integrations, bots and applications.
 
-The project is developed by a global community of contributors and is maintained and supported by a distributed company, [Mattermost Inc.](https://mattermost.com/about-us/), based in Palo Alto, California.
+The project is developed by a global community of contributors and is maintained and supported by a distributed company, {{< newtabref href="https://mattermost.com/about-us/" title="" >}}, based in Palo Alto, California.
```
Comment (neflyte) [inline site/content/contribute/more-info/team_contributions/_index.md:11]: This link needs a title.

File: site/content/contribute/more-info/server/rest-api.md [hunk 5]
```diff
@@ -28,18 +28,18 @@ To add an endpoint to API version 4, each item on the following checklist must b
...
-2. Find the `.yaml` file in the [/source/v4](https://github.com/mattermost/mattermost-api-reference/tree/master/v4/source) directory that fits your endpoint.
-    - For example, if you were adding the `GET /users/{user_id}` endpoint you would be looking for [users.yaml](https://github.com/mattermost/mattermost-api-reference/blob/master/v4/source/users.yaml)
-    - If the file doesn't exist yet, you might need to create it and update the [Makefile](https://github.com/mattermost/mattermost-api-reference/tree/master/Makefile) to include it
+2. Find the `.yaml` file in the {{< newtabref href="https://github.com/mattermost/mattermost-api-reference/tree/master/v4/source" title="/source/v4" >}} directory that fits your endpoint.
+    - For example, if you were adding the `GET /users/{user_id}` endpoint you would be looking for {{< newtabref href="https://github.com/mattermost/mattermost-api-reference/blob/master/v4/source/users.yaml" title="users.yaml" >}}
+    - If the file doesn't exist yet, you might need to create it and update the {{< newtabref href="https://github.com/mattermost/mattermost-api-reference/tree/master/Makefile" title="" >}} to include it
```
Comment (neflyte) [inline site/content/contribute/more-info/server/rest-api.md:42]: This link also needs a title.

**Review by neflyte — CHANGES_REQUESTED** [review]
Looks great! I found a few small issues, but we're ready to merge after that. :)

**Comment by ziriuz84** [comment]
I corrected typos and reverted js file

**Comment by ziriuz84** [comment]
Conflicts resolved

**Comment by cwarnermm** [comment]
@ziriuz84 - We can merge your PR into the codebase when you've resolved the merge conflicts on this PR. Thanks!

**Comment by ziriuz84** [comment]
Is there any other problem with the PR?

File: site/content/contribute/why-contribute/_index.md [hunk 6]
```diff
@@ -84,17 +86,19 @@ Good product and developer documentation content is as important as good code! I
...
+    
     -  You can get paid to write technical content for developers through the Mattermost {{< newtabref href="https://mattermost.com/blog/blog-announcing-community-writing-program/" title="community writing program" >}}.
-
     - If you see a problem with any webpages, blog posts, or other content on {{< newtabref href="https://mattermost.com" title="Mattermost.com" >}}, you can notify us via {{< newtabref href="https://community.mattermost.com/core/channels/mattermost-blog" title="the Content channel" >}} on the Mattermost Community Server.
+
     - Share your contributor or user experience! Mention us when you promote your work within our community, and we’ll amplify the message through Mattermost social media platforms.
     - Want to lead a social community? We can provide advice and resources to help you in the {{< newtabref href="https://community.mattermost.com/core/channels/community-team" title="Community channel" >}} on the Mattermost Community Server.
 
-## You want to make something more inclusive or accessible
+## You want to make something morDropdowne inclusive or accessible
```
Comment (neflyte) [inline site/content/contribute/why-contribute/_index.md:101]: I think this was unintended; could you please revert this change?

**Review by neflyte — CHANGES_REQUESTED** [review]
I found one small typo, but it looks great otherwise! :smile:

File: site/content/contribute/why-contribute/_index.md (hunk 6 above)
Comment (ziriuz84) [reply site/content/contribute/why-contribute/_index.md]: Sorry :sweat_smile: 
Resolved

**Review by neflyte — APPROVED** [review]
LGTM!

**Review by cwarnermm — APPROVED** [review]
